dj/construction/build.py

```diff
@@ -160,10 +160,15 @@ def _build_joins_for_dimension(
                 left_table.add_ref_column(
                     cast(ast.Column, join_left_columns[join_col.name]),
                 )
+                join_right_col = join_right_columns[
+                    join_col.dimension_column or "id"
+                ].copy()
+                join_right_col.name = join_right_col.alias_or_name
+                join_right_col.alias = None
                 join_on.append(
                     ast.BinaryOp.Eq(
                         join_left_columns[join_col.name],
-                        join_right_columns[join_col.dimension_column or "id"],
+                        join_right_col,
```

Comment:
Instead of making a copy of the column like you do a few lines above, could we just handle it inside of the `BinaryOp.__str__` function?

If it's because we only want to preference the alias in certain situations, maybe we can have a `prefer_alias: Optional[bool] = False` attribute on BinaryOp, and then do something like this in `BinaryOp.__str__` (while using `prefer_alias = True`on this line)
```py
    def __str__(self) -> str:
        if self.prefer_aliases and self.right.alias:
            right = self.right.copy()
            right.name = self.right.alias
            right.alias = None
        else:
            right = self.right
        ret = f"{self.left} {self.op.value} {right}"
        if self.parenthesized:
            return f"({ret})"
        return ret
```
Sort of a separate question: why do we only need to do this to the right column?

Reply:
Yeah, having this live in `__str__` makes more sense to me as well. I moved the alias promotion to a separate function on `Column`, since this can't apply to non-Column generic Expressions. And good call -- I added this check to both the left and the right column.